backend/src/plugins/Automod/actions/startThread.ts

```diff
@@ -0,0 +1,90 @@
+import { GuildFeature, ThreadAutoArchiveDuration } from "discord-api-types";
+import { TextChannel } from "discord.js";
+import * as t from "io-ts";
+import { renderTemplate, TemplateSafeValueContainer } from "src/templateFormatter";
+import { convertDelayStringToMS, tDelayString, tNullable } from "src/utils";
+import { savedMessageToTemplateSafeSavedMessage, userToTemplateSafeUser } from "src/utils/templateSafeObjects";
+import { noop } from "../../../utils";
+import { automodAction } from "../helpers";
+
+export const StartThreadAction = automodAction({
+  configType: t.type({
+    name: tNullable(t.string),
+    auto_archive: tNullable(t.number),
```

Comment:
Due to the way this is handled, we might want to consider making this a `tDelayString` that's validated to be one of the valid options.

Reply:
Done.
Not sure if this was the best approach calculations-wise, please check it out.